backend/src/impl/db_utils/system_db_utils.py

```diff
@@ -435,15 +432,20 @@ def db_operations(session: ClientSession) -> str:
                 system_id = DBUtils.insert_one(
                     DBUtils.DEV_SYSTEM_METADATA, document, session=session
                 )
-                # Compress the system output
+                # Compress the system output and upload to Cloud Storage
                 insert_list = []
                 sample_list = [general_to_dict(v) for v in system_output_data.samples]
-                sample_compressed = SystemDBUtils._compress_data(sample_list)
+
+                blob_name = f"{system_id}/{SystemDBUtils._SYSTEM_OUTPUT_CONST}"
```

Comment:
I'm not sure how the GCP file system works, but would it be OK to, for example, have 1M files in one directory? If so then this is fine, but if not then we might want to have a hierarchical structure such as `f"{system_id[0:3]}/{system_id}/{SystemDBUtils._SYSTEM_OUTPUT_CONST}"`

Reply:
I couldn't find a place in their [doc](https://cloud.google.com/storage/quotas#:~:text=There%20is%20no%20limit%20to%20the%20number%20of%20reads%20for,are%2C%20however%2C%20bandwidth%20limits.) where it explicitly says this but I think Cloud Storage (like S3) can basically hold any amount of data (S3 documentation says unlimited scalability). The only limit is that each object cannot exceed 5TB. So, I think this implementation is scalable.

re creating folders: Cloud Storage uses a flat namespace. If the object name contains "/"s, they are treated as part of the name. It does support [folders](https://cloud.google.com/storage/docs/folders) but it basically creates an illusion for the user so it's easier to navigate the files with `gsutil` or on the web UI.

Reply:
> Cloud Storage uses a flat namespace.

Correct. Cloud Storage (and some other storage services) adopts flat ("no folders") structure and all allowed characters (including `/`) are treated as a part of file name.

Cloud storage can store basically unlimited number of files as long as the data center allows. The only concern is the increase of cost of managing the data.